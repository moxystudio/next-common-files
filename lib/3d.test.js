'use strict';

const { with3D } = require('../index.js');

const options = {
    dev: true,
    isServer: true,
};

const createConfig = () => ({
    module: {
        rules: [],
    },
});

it('should configure a unique rule correctly', () => {
    const webpackConfig = with3D()().webpack(createConfig(), options);

    expect(webpackConfig.module.rules).toHaveLength(1);
    expect(webpackConfig.module.rules[0]).toMatchSnapshot();
});

it('should call nextConfig webpack if defined', () => {
    const nextConfig = {
        webpack: jest.fn(() => 'foo'),
    };

    const webpackConfig = with3D()(nextConfig).webpack(createConfig(), options);

    expect(nextConfig.webpack).toHaveBeenCalledTimes(1);
    expect(webpackConfig).toBe('foo');
});
