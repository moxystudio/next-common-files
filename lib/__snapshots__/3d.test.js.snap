// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should configure a unique rule correctly 1`] = `
Object {
  "loader": "<PROJECT_ROOT>/node_modules/url-loader/dist/cjs.js",
  "options": Object {
    "emitFile": false,
    "limit": 0,
    "name": "3d/[name].[ext]",
    "outputPath": "static/chunks/media",
    "publicPath": "/_next/static/chunks/media",
  },
  "test": /\\\\\\.\\(obj\\|mtl\\|fnt\\|glb\\)\\$/,
}
`;
